### PR TITLE
Split into versioned yaml files; update README.md.

### LINKED ISSUE #21: consider bumping pedantic by major versions when adding a lint
I've seen CIs for several projects break after the release of the latest `package:pedantic` - v`1.6.0`.

In the readme for this project, I see:

> This example uses a pinned version because every added lint rule could break a build for projects using it.

You might consider either
- making this more explicit - only recommend that consumers use specific versions, or pin to narrow ranges (`>=1.5.0 <1.6.0`).
- or, only make pedantic more restrictive - adding more rules - on upgrades to major versions. That would allow projects to use deps like `^1.0.0` and adjust to new versions of pedantic at a more leisurely pace. That would probably also follow the guidance of semver more closely.

### PR DESCRIPTION
**Pull request by davidmorgan**
I verified correct behaviour in IntelliJ when importing either the main or versioned options files.

Fixes #21

### CODE REVIEW

**Comment by davidmorgan** [comment]
Note to self: update https://dart.dev/guides/language/analysis-options when this is in.

File: README.md [hunk 1]
```diff
@@ -33,30 +33,39 @@ Here is how static analysis is used internally at Google:
 
 ## Enabled Lints
 
-The currently enabled lints can be found in the sample
-[analysis_options.yaml](https://github.com/dart-lang/pedantic/blob/master/lib/analysis_options.yaml).
+The currently enabled lints can be found in
+[analysis_options.1.7.0.yaml](https://github.com/dart-lang/pedantic/blob/master/lib/analysis_options.1.7.0.yaml).
```
Comment (devoncarew) [inline README.md]: Here, do you want to reference a specific version, or just the main `analysis_options.yaml`?

File: lib/analysis_options.yaml [hunk 2]
```diff
@@ -4,31 +4,10 @@
...
-    - prefer_is_not_empty
-    - recursive_getters
-    - slash_for_doc_comments
-    - type_init_formals
-    - unawaited_futures
-    - unnecessary_null_in_if_null_operators
-    - unrelated_type_equality_checks
-    - use_rethrow_when_possible
-    - valid_regexps
+include: package:pedantic/analysis_options.1.6.0.yaml
```
Comment (devoncarew) [inline lib/analysis_options.yaml]: You probably mean `1.7.0` here?

Comment (davidmorgan) [reply lib/analysis_options.yaml]: Done.

File: README.md (hunk 1 above)
Comment (davidmorgan) [reply README.md]: I want the specific version so that people can click through and see the lints.

File: lib/analysis_options.yaml (hunk 2 above)
Comment (davidmorgan) [reply lib/analysis_options.yaml]: Done.

**Comment by davidmorgan** [comment]
Thanks!

Copying rather than importing is a possibility; the main downside I see is that I'd like to have a different note at the top of `analysis_options.yaml` warning people that it's not pinned. So they'd have to be slightly different rather than a copy.

I tried this setup in IntelliJ and didn't see any problems, so I'm not worried about performance.

Let's give this a try and see how it works :)

**Comment by devoncarew** [comment]
> I tried this setup in IntelliJ and didn't see any problems, so I'm not worried about performance.

I have seen issues with source bases the size of flutter, when there were several chained analysis options files. I don't have the specific PR, but it was in the nature of a 25%-50% performance regression to analyze the entire flutter repo.

I'd avoid multiple chained references until we have a high confidence that it doesn't have a performance impact.

**Comment by davidmorgan** [comment]
Thanks Devon.

I did some investigation to try to repro what you're describing, but didn't find anything. If you have any suggestions for further things to try, I'm interested in tracking down and fixing, since unexplained slowdowns are bad for the dev cycle :)

Here's what I tried:

 - Create a project with ~10,000 src files to analyze (just copies of the same file)
 - Create a chain of 1000 `analysis_options.yaml` files, with the real options at the end
 - Benchmark three cases: with direct `analysis_options.yaml`, with a single `import`, and with the chained 1000 imports; and benchmark using `dartanalyzer` on the command line, and in IntelliJ.

Results showed no appreciable difference however the analysis options were supplied. For the command line:

analyze with direct: 65.44s, 65.42s, 67.01s, 66.93s, 65.34s
analyze with pedantic 1.6.0: yaml: 67.64s, 67.13s, 68.34s, 65.68s, 67.26s
analyze with 1000 chained yaml files: 67.45s, 67.60s, 66.79s, 67.95s, 67.18s

and analysis in IntelliJ was about 42s in all three cases. When benchmarking in IntelliJ I had to delete ~/.dartServer/analysis-driver to clear caches. Measurements in IntelliJ were approximate as I had no way to record the exact time.

**Comment by devoncarew** [comment]
The context for my concerns w/ the linking approach is:

- https://github.com/flutter/flutter/pull/26244
- and, https://github.com/dart-lang/sdk/issues/35637

**Comment by davidmorgan** [comment]
Thanks :)

**Pull request merged**